packages/teleport/src/config.ts

```diff
@@ -65,7 +66,7 @@ const cfg = {
     consoleNodes: '/web/cluster/:clusterId/console/nodes',
     consoleConnect: '/web/cluster/:clusterId/console/node/:serverId/:login',
     consoleSession: '/web/cluster/:clusterId/console/session/:sid',
-    player: '/web/cluster/:clusterId/session/:sid',
+    player: '/web/cluster/:clusterId/session/:sid', // ?recordingType=ssh|desktop
```

Comment:
i think you can do something like below: https://github.com/remix-run/react-router/blob/v3/docs/guides/RouteMatching.md#path-syntax
```suggestion
    player: '/web/cluster/:clusterId/session/:sid(?recordingType=:recordingType)'
```

Reply:
That guide is for an older version of React Router (v3), we are using v5.1.1. In the case of your suggestion, the parentheses and "?" are interpreted as regex symbols. I've done some more research, and afaict React Router doesn't have much if any support for dealing with query parameters.

https://stackoverflow.com/a/35604855/6277051
https://learnwithparam.com/blog/how-to-handle-query-params-in-react-router/

Reply:
Just for a reference if you're interested, my go to for defining and typing query params is to use [use-query-params](https://www.npmjs.com/package/use-query-params)

Reply:
@hatched thanks, looks nice enough but my first impression is that its a bit overpowered for this application. We're not using them for any sort of complex state management.